maven-surefire-plugin/src/site/apt/examples/fork-options-and-parallel-execution.apt.vm

```diff
@@ -181,6 +181,67 @@ Fork Options and Parallel Test Execution
   See the keywords: <volatile>, <synchronized>, <<immutable>> and <final> in
   {{{https://jcp.org/en/jsr/detail?id=133}Java Memory Model - JSR-133}}.
 
+* Parallel Test-Suite Execution
+
+  ${project.artifactId}'s notion of "suite" is related to
+  {{{https://github.com/junit-team/junit4/wiki/Aggregating-tests-in-suites}junit4 Suite}}. 
+
+  For example, say your tests are laid out like this:
+
++---+
+src/test/java
++-features.areaA
+| +-SomeStory.java
+| +-AnotherStory.java
+| ...
++-features.areaB
+| +-DifferentStory.java
+| +-OtherStory.java
+| ...
+...
++---+
+
+  You would add a TestSuite.java for each package, that would look something like:
+
++---+
+package features.areaA;
+
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+
+@RunWith(Suite.class)
+@Suite.SuiteClasses({
+    SomeStory.class,
+    AnotherStory.class,
+    ...
+})
+public class TestSuite {
+    /* Suite holder */
+}
++---+
+
+  and similarly for <<<package features.areaB>>>. You would then configure
+  ${project.artifactId} like so:
+
++---+
+<plugin>
+  <groupId>${project.groupId}</groupId>
+  <artifactId>${project.artifactId}</artifactId>
+  <version>${project.version}</version>
+  <configuration>
+    <includes>
+      <include>features.*.*Suite</include>
+    </includes>
+    <parallel>suites</parallel>
+    <threadCount>4</threadCount>
```

Comment:
What happens with `3` or `2`. Did you make an experiment?

Reply:
Well, my explanation below this is "This would then run **up to** 4 threads" (added emphasis). So I would hope that people can deduce that using 3 or 2, would "then run up to 3 or 2, respectively, threads.

Reply:
I wrote the `ParallelComputer` for Surefire. But it was in 2013. Now I do not exactly remember whether the suites are searched recursively of they are flat. If the search is recursive then both packages and their content would run in parallel completely.

Therefore I was asking fot the number 4 because this sounds to me like all is parallel.

Reply:
@rehevkor5 
After seen the code in `ParallelComputerBuilder` the search is not recursive and searches the runners only in two levels. Therefore you should use `2` and not `4`. Then the users would understand it. They can use another number but the effect would be different. You can explain this in the document because then the users would understand what they may expect.

Reply:
Could you please explain what you mean. Especially "use another number but the effect would be different". I do not understand.
In my actual use-case I am running into problems where a bunch of tests pass and a bunch fail, and I do not know why.

Reply:
Also, am I correct in that, in this specific case, `threadCount=X` would have the exact same effect as `threadCountSuites=X`?
Should I add that to the docs here?

Reply:
> Also, am I correct in that, in this specific case, `threadCount=X` would have the exact same effect as `threadCountSuites=X`?
> Should I add that to the docs here?

You'r right, it has identical effect **in this case**.

Reply:
> Could you please explain what you mean. Especially "use another number but the effect would be different". I do not understand.
> In my actual use-case I am running into problems where a bunch of tests pass and a bunch fail, and I do not know why.

I wanted to say that Surefire nested suites in third level are not running in parallel.
Your example is okay, it has package `featureA` and it means that `featureB` is concurrent. But I want to say that our users should not expect that a nested package `featureA.pkg1` and `featureA.pkg2` would be concurrent. The Surefire would run `pkg1` and `pkg2` in a sequence.  I guess you know what I mean. If somebody expects making them parallel then he would use 4 threads or 8 threads but I think it would not be... So hopefully your case works but the users should not expect any generic approach with infinite depth of parallel suites.

Reply:
> Well, my explanation below this is "This would then run **up to** 4 threads" (added emphasis). So I would hope that people can deduce that using 3 or 2, would "then run up to 3 or 2, respectively, threads."

If I understand you correctly, you want to achive the following. You want to run package `featureA` in thread T1, and `featureB` in thread T2. Am I right?

Reply:
In a word: yes.

Reply:
So it means that there would be no difference between two and four threads, because there are two packages and there would still only two parallel executions and not four. Why you write 4 in the documentation? If the users has a reason, he would change it, he can setup parallel classes to but it is up to him - not us.

Reply:
done

Reply:
I am trying to figure out why my tests keep failing.
Right now, just to figure out what is going on, I have 4 *Suite.java, and I am using threadCountSuites=2. Please note that my tests are using Selenium. When I run my tests, it looks like 4 browsers (tests) are launched.
If I change the number of *Suite.java files, then I get the same 4 browsers.
However, if I change the number of threads, for example threadCountSuites=4, I get twice as many browsers launched.
What am I doing wrong?